### ISSUE TITLE
Can not reproduce the experimental results

### ISSUE THREAD

**Issue by EigenvectorOfFate** [thuml/RoPINN#1]
I appreciate that the authors provide the source codes with many variations of PINNs.

I have tried several combinations for solving the 1d-reaction, 1d-wave, and 1d-convection equations. However, I found that there was something wrong with the experimental results. For example, the exact u(x,t) of 1d-reaction is shown in the following figure.

[1dreaction_exact.pdf](https://github.com/user-attachments/files/17641599/1dreaction_exact.pdf)

After running python 1d_reaction_point_optimization.py --model PINN, I got the following results:
Loss Res: 0.010579, Loss_BC: 0.000008, Loss_IC: 0.197978
Train Loss: 0.208565
relative L1 error: 0.993095
relative L2 error: 0.990091

[1dreaction_PINN_point_optimization_pred.pdf](https://github.com/user-attachments/files/17641613/1dreaction_PINN_point_optimization_pred.pdf)

After running python 1d_reaction_region_optimization.py --model PINN, I got the following results:

Loss Res: 0.010603, Loss_BC: 0.000009, Loss_IC: 0.197934
Train Loss: 0.208547
relative L1 error: 0.992901
relative L2 error: 0.989975

[1dreaction_PINN_region_optimization_pred.pdf](https://github.com/user-attachments/files/17641619/1dreaction_PINN_region_optimization_pred.pdf)

It can be seen that a relatively accurate approximate solution is not obtained, and similar situations occur for the 1d wave and 1d convection equations.

**Comment by EigenvectorOfFate** [comment]
I also tried to use the source code to solve the 1d Allen-Cahn equation with the following code:
`for i in tqdm(range(1000)):
    def closure():
        pred_res = model(x_res, t_res)
        pred_left = model(x_left, t_left) 
        pred_upper = model(x_upper, t_upper) 
        pred_lower = model(x_lower, t_lower) 
        
        u_x = \
            torch.autograd.grad(pred_res, x_res, grad_outputs=torch.ones_like(pred_res),
                                retain_graph=True,
                                create_graph=True)[0]
        u_xx = torch.autograd.grad(u_x, x_res, grad_outputs=torch.ones_like(pred_res), retain_graph=True,
                                   create_graph=True)[0]
        u_t = \
            torch.autograd.grad(pred_res, t_res, grad_outputs=torch.ones_like(pred_res),
                                retain_graph=True,
                                create_graph=True)[0]
        bnd_upper = torch.autograd.grad(pred_upper, x_upper, grad_outputs=torch.ones_like(pred_upper), retain_graph=True,
                                create_graph=True)[0]
        bnd_lower = torch.autograd.grad(pred_lower, x_lower, grad_outputs=torch.ones_like(pred_lower), retain_graph=True,
                                create_graph=True)[0]
        
        loss_res = torch.mean((u_t - p * pred_res - r * pred_res**3 - q * u_xx) ** 2)
        loss_bc = torch.mean((bnd_upper - bnd_lower) ** 2)
        loss_ic = torch.mean((pred_left[:, 0] - 0.25 * torch.sin(x_left[:, 0])) ** 2)

        loss_track.append([loss_res.item(), loss_bc.item(), loss_ic.item()])

        loss = loss_res + loss_bc + loss_ic
        optim.zero_grad()
        loss.backward()
        return loss
`

The expected approximate solution is

 <img width="295" alt="approximate_solution" src="https://github.com/user-attachments/assets/b042166c-587c-4038-8ab1-63d663619843">.

However, with the above code, I got the wrong numerical results.

<img width="297" alt="predicted _solution_using_the_above_code" src="https://github.com/user-attachments/assets/95903456-50f3-4cb7-a049-cdf165ef7a3e">

**Comment by wuhaixu2016** [comment]
Hi, thanks for your usage. 

(1) About RoPINN's performance in 1D Reaction.

According to my experiment, the reproduced results will be better than those reported in our paper. You can obtain my training loss log here:

<img width="835" alt="截屏2024-11-06 13 57 38" src="https://github.com/user-attachments/assets/3dc9537a-4da2-46cd-af28-9e110d218f61">

The final prediction error is like this: 

[1dreaction_PINN_region_optimization_error.pdf](https://github.com/user-attachments/files/17642386/1dreaction_PINN_region_optimization_error.pdf)

Here is my checkpoint. You can try to test the model with this checkpoint:

https://cloud.tsinghua.edu.cn/f/90e6cc1085f84cd9be6b/

(2) As for the 1d Allen-Cahn equation, we do not experiment with this problem in our paper but thanks for sharing this problem. I will try this task. If I have new results, I will update them here.

However, I think the results you provided have a clear mismatch axis. Maybe the current huge gap is caused by mismatched x and t axes during plotting.

**Comment by wuhaixu2016** [comment]
Besides, I have also tested the model performance in 1D wave and Convection. You can obtain my checkpoints for PINN region optimization here: https://cloud.tsinghua.edu.cn/d/d37aea18c1cb48038978/

Here are the training logs, where the reproduced results are quite close or even better than our paper reported. 

<img width="731" alt="截屏2024-11-06 14 08 26" src="https://github.com/user-attachments/assets/13cbe4f8-51e0-46fc-a545-ea7eb0276704">

<img width="813" alt="截屏2024-11-06 14 09 34" src="https://github.com/user-attachments/assets/53a8b8d1-6b73-43d1-919f-0d14a211aadb">

I think one potential question is the experimental environment. Please try this configuration: PYTORCH1.13.0 CUDA11.7

**Comment by EigenvectorOfFate** [comment]
I upgrated PyTorch version from 1.11.0 to 1.13.0 and got the right experimental results on the 1d-reaction, 1d-wave, and 1d-convection equations.
Besides, with the help of the region optimization paradigm, more accurate numerical results is obtained on the allen-cahn equation compared to the vanilla PINN. (Loss Res: 0.000016, Loss_BC: 0.000000, Loss_IC: 0.000006 Train Loss: 0.000022)
<img width="297" alt="截屏2024-11-06 15 20 19" src="https://github.com/user-attachments/assets/981f46ae-a3cb-4451-ac87-2fa058378828">
Thank you very much!

**Comment by wuhaixu2016** [comment]
Cool, thanks for your information. Glad to see the performance on 1d Allen-Cahn equation~
